File: Java/src/main/java/com/gildedrose/Items/BackstagePassItem.java
package com.gildedrose.Items;

import com.gildedrose.Item;

public class BackstagePassItem extends GildedItem {
    public BackstagePassItem(Item item) {
        super(item);
    }

    @Override
    protected void updateItemsPreSellInDecrement() {
        incrementItemQuality();
        if (item.sellIn < 11) {
            incrementItemQuality();
        }

        if (item.sellIn < 6) {
            incrementItemQuality();
        }
    }

    @Override
    protected void updateitemsPostSellInDecrement() {
        if (item.sellIn >= 0) {
            return;
        }

        item.quality = 0;
    }
}

File: go/gildedrose/gildeditems/backstage_pass_item.go
package gildeditems

type BackstagePass struct {
	*Item
}

func (item *BackstagePass) DoItemUpdate() {
	item.incQuality()
	if item.SellIn <= 10 {
		item.incQuality()
	}

	if item.SellIn <= 5 {
		item.incQuality()
	}
	item.SellIn--
	if item.SellIn < 0 {
		item.Quality = 0
	}
}
